lib/client/checks/http/http-executor.ts

```diff
@@ -0,0 +1,90 @@
+import * as logger from '../../../log';
+import * as version from '../../../version';
+import { axiosInstance } from '../../../axios';
+import { retry } from '../../retry/exponential-backoff';
+import type { AxiosResponse } from 'axios';
+import type { CheckId, CheckResult, CheckStatus } from '../types';
+
+export async function executeHttpRequest(
+  checkOptions: {
+    id: CheckId;
+    name: string;
+  },
+  httpOptions: {
+    url: string;
+    method: 'GET' | 'POST';
+    timeoutMs: number;
+  },
+): Promise<CheckResult> {
+  logger.debug({ checkId: checkOptions.id }, 'executing http check');
+
+  try {
+    const response = await retry<CheckResult>(
+      () =>
+        axiosInstance.request({
+          method: httpOptions.method,
+          url: httpOptions.url,
+          headers: {
+            'User-Agent': `broker client/${version} (http check service)`,
+          },
+          timeout: httpOptions.timeoutMs,
+          validateStatus: () => true,
+        }),
+      { retries: 3, operation: `http check ${checkOptions.id}` },
+    );
+    logger.trace({ response: response.data }, 'http check raw response data');
+
+    const checkResult = convertAxiosResponseToCheckResult(
+      {
+        id: checkOptions.id,
+        name: checkOptions.name,
+        url: httpOptions.url,
+        method: httpOptions.method,
+      },
+      response,
+    );
+
+    logger.debug(
+      { checkId: checkOptions.id, status: checkResult.status },
+      'completed http check execution',
+    );
+    return Promise.resolve(checkResult);
+  } catch (error) {
+    const errorMessage = `Error executing check with checkId ${checkOptions.id}`;
+    logger.debug({ error }, errorMessage);
+    throw new Error(errorMessage);
+  }
+}
+
+export function convertAxiosResponseToCheckResult(
+  check: { id: CheckId; name: string; url: string; method: 'GET' | 'POST' },
+  response: AxiosResponse,
+): CheckResult {
+  let status: CheckStatus;
+  if (response.status >= 200 && response.status <= 299) {
+    status = 'passing';
```

Comment:
I think this is fine for now but more elaborate conditions will be needed. 2xx response won't allow us to verify some more elaborate checks.

Reply:
Agree, let's add default statuses in the next iteration with possibility to override it if needed.